mindmeld/components/_config.py

```diff
@@ -522,6 +514,12 @@
     },
 }
 
+# A language-specific stemmer and tokenizer will be selected during runtime.
```

Comment:
Related to rivtik's point, but I would like to see the defaults for english here

Reply:
Below is what the default EN Config would look like based on changes in progress. I'm thinking about the best way to handle these defaults and it may make sense to have language specific normalizer lists in the file `EN_NORMALIZERS` and keep the other elements "dynamic" such that they are inferred based on the language. Ie. For tokenizer, it will select a SpacyTokenizer if the language is supported by Spacy, else, WhiteSpaceTokenier. Similar procedure with Stemmer. Thoughts?

```
DEFAULT_EN_TEXT_PREPARATION_CONFIG = {
    "preprocessors": [],
    "tokenizer": "SpacyTokenizer",
    "normalizers": [
        'RemoveAposAtEndOfPossesiveForm',
        'RemoveAdjacentAposAndSpace',
        'RemoveBeginningSpace',
        'RemoveTrailingSpace',
        'ReplaceSpacesWithSpace',
        'ReplaceUnderscoreWithSpace',
        'SeparateAposS',
        'ReplacePunctuationAtWordStartWithSpace',
        'ReplacePunctuationAtWordEndWithSpace',
        'ReplaceSpecialCharsBetweenLettersAndDigitsWithSpace',
        'ReplaceSpecialCharsBetweenDigitsAndLettersWithSpace',
        'ReplaceSpecialCharsBetweenLettersWithSpace',
        'RemoveSpecialCharsBeforePipe',
        'ReplaceEndBracketAndFollowingSpecialCharsBeforeSWithSpace'
        'Lowercase',
        'ASCIIFold'
    ],
    "stemmer": "EnglishNLTKStemmer"
}
```

Reply:
If we place English defaults here then the correct (dynamically created) Tokenizers and Stemmers won't be expected. Since a dynamic Tokenizer/Stemmer is only selected if the given attribute is None

Reply:
That does make sense if we can dynamically select the components.

Although, is it not possible to overwrite the default config here with the dynamic values? Not sure how that affects runtime but one idea could be to load the config from file or default to this, and then create a runtime config based on language selected. If nothing is obtained from the latter, then continue as is or otherwise overwrite the default.

Reply:
One of the reasons I mentioned default EN config was that most of the other functionalities in MM currently default to that. So this would be consistent, but depending upon consensus we can go the dynamic route as well.

Reply:
I think we should avoid having configs overwritten internally as that creates confusion to the user. To your point regarding our use of a default EN config elsewhere in the code: We can make a default en config and then have a check that goes (if language ==“en” return default_en_config). Such logic would be a bit redundant, however, it could be more consistent?

Reply:
After discussing this with Vijay, we decided to use the existing Normalization rules for English for all languages. If a user would like to update the rule set, they can paste a subset of the default rules into the 'normalizer' section of the config. These changes have been made